.github/workflows/smoke_test.yaml

```diff
@@ -25,6 +25,7 @@ jobs:
       run: |
         python -m pip install --upgrade pip setuptools wheel
         python -m pip install './funcx_sdk[test]'
+        python -m pip install './funcx_endpoint'
```

Comment:
Can we flag this line with a fixme comment to remove this install step in the future?

`funcx` should not require `funcx-endpoint`, since `funcx-endpoint` requires `funcx`. The codependence of these packages points at a need to either merge them into a single package or fix the use of `funcx-endpoint` components in `funcx`.